client/my-sites/checkout/checkout/checkout.jsx

```diff
@@ -341,19 +343,46 @@ export class Checkout extends React.Component {
 		if ( this.props.isNewlyCreatedSite && receipt && isEmpty( receipt.failed_purchases ) ) {
 			const siteDesignType = get( selectedSite, 'options.design_type' );
 			const hasGoogleAppsInCart = cartItems.hasGoogleApps( cart );
+			const hasConciergeSessionInCart = cartItems.hasConciergeSession( cart );
 
+			// Handle the redirect path after a purchase of GSuite or Concierge Session
 			// The onboarding checklist currently supports the blog type only.
-			if ( hasGoogleAppsInCart && domainReceiptId && 'store' !== siteDesignType ) {
+			if (
+				( hasConciergeSessionInCart || ( hasGoogleAppsInCart && domainReceiptId ) ) &&
+				'store' !== siteDesignType
+			) {
 				analytics.tracks.recordEvent( 'calypso_checklist_assign', {
 					site: selectedSiteSlug,
 					plan: 'paid',
 				} );
+
+				if ( config.isEnabled( 'upsell/concierge-session' ) && hasConciergeSessionInCart ) {
+					return `/checklist/${ selectedSiteSlug }`;
+				}
 				return `/checklist/${ selectedSiteSlug }?d=gsuite`;
 			}
 
-			if ( ! hasGoogleAppsInCart && cartItems.hasDomainRegistration( cart ) ) {
+			if (
```

Comment:
The flow you've explained is correct. I only mentioned `don't choose "Sell products or collect payments"...` to make sure the outcome of the testing steps was consistent in that it redirected to the checklist after purchase.

Reply:
> So, should we be showing the upsell page for store selection?

Yes

> If yes, should we be taking them to the checklist page instead of Thank You page?

We're sticking with the current checklist logic for now, which is to not show the checklist for store selection.